source/agora/test/MultiRoundConsensus.d

```diff
@@ -0,0 +1,149 @@
+/*******************************************************************************
+
+    Tests for reaching consensus in multiple rounds instead of 1 round.
+    In this test, we make nodes reject nominations for several rounds
+    deliberately until one is accepted at a round R, where R could be arbitrarily
+    high.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.MultiRoundConsensus;
+
+version (unittest):
+
+import agora.api.Validator;
+import agora.common.Config;
+import agora.common.Task;
+import agora.consensus.data.Block;
+import agora.consensus.data.ConsensusParams;
+import agora.consensus.protocol.Nominator;
+import agora.common.crypto.Key;
+import agora.common.Hash;
+import agora.consensus.data.Transaction;
+import agora.consensus.data.genesis.Test;
+import agora.network.Clock;
+import agora.network.NetworkManager;
+import agora.node.Ledger;
+import agora.test.Base;
+
+import core.stdc.inttypes;
+import core.stdc.time;
+import core.thread;
+
+import geod24.Registry;
+
+import scpd.types.Stellar_types;
+import scpd.types.Stellar_SCP;
+
+/// ditto
+unittest
+{
+    static class CustomNominator : TestNominator
```

Comment:
```suggestion
    extern(C++) static class CustomNominator : TestNominator
```

You will also need to change `static shared` into `__gshared static` for the `round_number`

Reply:
I will apply `__gshared static` to the code. But is the `extern(C++)` needed although the code as follows exists?
[extern (C++)](https://github.com/linked0/agora/commit/8f9305c4a1f46b3116e8facd6bf89085d0c138ca#diff-99dfbae9b7bf38c31b60a42b7390c181R63-R65)

Reply:
I've had segfaults when I forgot to mark the class itself as `extern(C++)`, so I recommend not risking by mixing D and C++ class linkage. There may be undiscovered compiler bugs with it, but I haven't reduced it to a test-case yet.

Reply:
> I've had segfaults when I forgot to mark the class itself as extern(C++), so I recommend not risking by mixing D and C++ class linkage.

OK, I got it. Thanks.

Reply:
> You will also need to change static shared into __gshared static for the round_number

There is only one copy of such a variable in the entire program and all threads can share that variable. __gshared is necessary when interacting with libraries of languages like C and C++ where data sharing is automatic by default.
- Cehreli, Ali. Programming in D: Tutorial and Reference (p. 636). Ali Cehreli. Kindle Edition.

Reply:
More specifically:

```D
extern(C++) class C
{
    shared static int x;
}
```

$ dmd test.d
> test.d(3): Error: variable test.C.x Internal Compiler Error: C++ static non-__gshared non-extern variables not supported